staging/src/k8s.io/apiextensions-apiserver/test/integration/apiapproval_test.go

```diff
@@ -72,8 +72,8 @@ func TestAPIApproval(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	err = wait.PollImmediate(100*time.Millisecond, 30*time.Second, func() (bool, error) {
-		approvedKubeAPI, err = apiExtensionClient.ApiextensionsV1().CustomResourceDefinitions().Get(context.TODO(), approvedKubeAPI.Name, metav1.GetOptions{})
+	err = wait.PollUntilContextTimeout(context.Background(), 100*time.Millisecond, 30*time.Second, true, func(ctx context.Context) (bool, error) {
+		approvedKubeAPI, err = apiExtensionClient.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, approvedKubeAPI.Name, metav1.GetOptions{})
```

Comment:
While on it: use a dedicated `err`, i.e. with `:=`. Otherwise this style risks a race.

Reply:
Fixed it https://github.com/kubernetes/kubernetes/commit/ed7f7ce92b06b2452a3d879b4c70d34f5d31ff23